### PR TITLE
Added a Santa class that extends Character

### PR DESCRIPTION
**Pull request by M1tchM355**

### CODE REVIEW

File: src/Santa.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    public int getEnergy() {
+        return energy;
+    }
+
+    public void attack(Character target) {
+        double explodeChance = Math.random();
+        if (energy < 10) {
+            System.out.println(getName() + "is too tired, so they fall asleep");
+            try {
+                Thread.sleep(10000);
```
Comment (amikhalev) [inline src/Santa.java]: You're going to be blocking the whole game.... that's ok I guess

File: src/Santa.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            System.out.println(target.getName() + "already has full health! There were no effects");
+        } else {
+            if (explodeChance > .85) {
+                System.out.println(getName() + "threw a present at " + target.getName()
+                        + ", and it exploded green healing goo! 30 health was added!");
+                if (target.health > 70) {
+                    target.health = 100;
+                } else {
+                    target.health += 30;
+                }
```
Comment (amikhalev) [inline src/Santa.java]: Nice edge case checking! Although I might do it like
```java
target.health += 30;
if (target.health > 100) {
    target.health = 100;
}
```
so you don't have to worry about updating the `70` if you update the `30`

File: src/Santa.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+            } else {
+                System.out.println(getName() + "threw a present at " + target.getName() + ", and it added 10 health!");
+                if (target.health > 90) {
+                    target.health = 100;
+                } else {
+                    target.health += 10;
+                }
```
Comment (amikhalev) [inline src/Santa.java]: Same here

File: src/Santa.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+            if (explodeChance > .85) {
+                System.out.println(getName() + "threw a present at " + target.getName()
+                        + ", and it exploded green healing goo! 30 health was added!");
+                if (target.health > 70) {
+                    target.health = 100;
+                } else {
+                    target.health += 30;
+                }
+            } else {
+                System.out.println(getName() + "threw a present at " + target.getName() + ", and it added 10 health!");
```
Comment (amikhalev) [inline src/Santa.java]: Can you make this line a bit shorter by inserting some newlines? Or autoformat in intellij should do it (Ctrl+Shift+A, then type "format", hit enter)

File: src/Present.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
+/**
+ * Created by messy on 12/12/2016.
```
Comment (fortninja) [inline src/Present.java]: Put some useful information here instead of this default comment

File: src/Present.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+                    target.health += 10;
+                }
+            }
+        }
+        santa.setEnergy(santa.getEnergy()-10);
+        numOfPresents -= 1;
+        return true;
+    }
+
+    public void beAddedToSanta(Santa owner) {
```
Comment (fortninja) [inline src/Present.java]: This method is not necessary

File: src/Present.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+        santa = owner;
+    }
+
+    public int getNumOfPresents() {
+        return numOfPresents;
+    }
+
+    void setNumOfPresents(int numOfPresents) {
```
Comment (fortninja) [inline src/Present.java]: set this to public

File: src/Santa.java [hunk 8]
```diff
@@ -0,0 +1,74 @@
+import java.util.List;
+
+/**
+ * Created by Mitch Messerly on 12/6/2016.
```
Comment (fortninja) [inline src/Santa.java]: put some info here

File: src/Santa.java [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+        return energy;
+    }
+
+    public void describeSelf() {
+        System.out.println("My name is " + getName() + ", their suit is " + getSuitColor() + ", they have "
+                + ((Present) weapons.get(0)).getNumOfPresents() + " presents, they are being followed by " + getNumOfElves()
+                + "elves, and they have " + getEnergy() + "energy");
+    }
+
+    public String getPrimaryWeapon() {
```
Comment (fortninja) [inline src/Santa.java]: this method should be removed

**Pull request merged**